File: Coding Interviews/03 - Coding Quality/E373.PartitionArraybyOddandEven.java
public class Solution {
    /*
     * @param nums: an array of integers
     * @return: nothing
     */
    public void partitionArray(int[] nums) {
        // write your code here
        int l = 0, r = nums.length - 1;
        while (l < r) {
            while (l < r && nums[l] % 2 == 1) l++;
            while (l < r && nums[r] % 2 == 0) r--;

            if (l < r) {
                int tmp = nums[l];
                nums[l++] = nums[r];
                nums[r--] = tmp;
            }
        }
    }
}

File: Coding Interviews/03 - Coding Quality/E373.PartitionArraybyOddandEven.py
class Solution:
    """
    @param: nums: an array of integers
    @return: nothing
    """
    def partitionArray(self, nums):
        # write your code here
        if not nums:
            return nums

        l, r = 0, len(nums) - 1
        while l <= r:
            if nums[l] % 2 == 1:
                l += 1
            elif nums[r] % 2 == 0:
                r -= 1
            else:
                nums[l], nums[r] = nums[r], nums[l]
                l += 1
                r -= 1
